pkg/userdata/flatcar/provider.go

```diff
@@ -170,7 +172,7 @@ const userDataIgnitionTemplate = `passwd:
         {{end}}
 {{- end }}
 
-{{- if .ProviderSpec.Network }}
+{{- if .StaticIPConfigProvided }}
```

Comment:
You could just `.ProviderSpec.Network.IsStaticIPConfig`

like in https://stackoverflow.com/a/10200273

Reply:
Forgot about this completely 😁